goal/src/main/java/org/openapitools/vertxweb/server/api/impl/PetsApiImpl.java

```diff
@@ -35,6 +42,64 @@ public PetsApi listPetsHandler(Handler<RoutingContext> handler) {
         return this;
     }
 
+    @Override
+    public <T> PetsApi listPets(Function<ListPetsArguments, Future<T>> function) {
+        builder.operation("showPetById").handler(ctx -> {
+            // TODO: extract the function to handler into a helper as it must be reused
+            try {
+                function
+                        .apply(new ListPetsArguments() {
+                            @Override
+                            public RoutingContext context() {
+                                return ctx;
+                            }
+
+                            @Override
+                            public Long limit() {
+                                return ctx.get("limit");
+                            }
+                        })
+                        .onFailure(ctx::fail)
```

Comment:
From here down it's just a bad copy paste from the `respond` method, this can be simplified by using the read respond